tests/ui/self-profile/pretty_print_no_ice.rs

```diff
@@ -0,0 +1,14 @@
+// Checks that when we use `-Zself-profile-events=args`, it is possible to pretty print paths
+// using `trimmed_def_paths` even without producing diagnostics.
+//
+// Issue: <https://github.com/rust-lang/rust/issues/144457>.
+
+//@ compile-flags: -Zself-profile -Zself-profile-events=args
```

Comment:
I believe this is going to put the `.mm_profdata` files in the root and not in the temporary build test paths.

Reply:
Uhh, I'm used to run-make tests that have properly scoped working directories. You're right. Do you know if there's a way in the test header to use some temporary dir? It should be enough to say `-Zself-profile=foo`.

Reply:
Ah, got it, `{{build-base}}`.